File: db4ocs/test/src/com/db4o/test/CascadeOnActivate.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.test;

import com.db4o.*;
import com.db4o.config.*;
import com.db4o.ext.*;
import com.db4o.query.*;

import db4ounit.*;
import db4ounit.extensions.*;

public class CascadeOnActivate extends AbstractDb4oTestCase {

	public String name;

	public CascadeOnActivate child;

	protected void configure(Configuration config) {
		config.objectClass(this).cascadeOnActivate(true);
	}

	public void store(ExtObjectContainer oc) {
		name = "1";
		child = new CascadeOnActivate();
		child.name = "2";
		child.child = new CascadeOnActivate();
		child.child.name = "3";
		oc.set(this);
	}

	public void conc(ExtObjectContainer oc) {
		Query q = oc.query();
		q.constrain(CascadeOnActivate.class);
		q.descend("name").constrain("1");
		ObjectSet os = q.execute();
		CascadeOnActivate coa = (CascadeOnActivate) os.next();
		CascadeOnActivate coa3 = coa.child.child;
		Assert.areEqual("3", coa3.name);
		oc.deactivate(coa, Integer.MAX_VALUE);
		Assert.isNull(coa3.name);
		oc.activate(coa, 1);
		Assert.areEqual("3", coa3.name);
	}

	public void concIndexed(ExtObjectContainer oc) {
		oc.configure().objectClass(CascadeOnActivate.class).objectField("name")
				.indexed(true);
		conc(oc);
	}

}

File: db4on/test/src/com/db4o/db4ounit/common/querying/CascadeOnActivate.cs
namespace com.db4o.db4ounit.common.querying
{
	public class CascadeOnActivate : Db4oUnit.Extensions.AbstractDb4oTestCase
	{
		public string name;

		public com.db4o.db4ounit.common.querying.CascadeOnActivate child;

		protected override void Configure(com.db4o.config.Configuration conf)
		{
			conf.ObjectClass(this).CascadeOnActivate(true);
		}

		protected override void Store()
		{
			com.db4o.db4ounit.common.querying.CascadeOnActivate coa = new com.db4o.db4ounit.common.querying.CascadeOnActivate
				();
			coa.name = "1";
			coa.child = new com.db4o.db4ounit.common.querying.CascadeOnActivate();
			coa.child.name = "2";
			coa.child.child = new com.db4o.db4ounit.common.querying.CascadeOnActivate();
			coa.child.child.name = "3";
			Db().Set(coa);
		}

		public virtual void Test()
		{
			com.db4o.query.Query q = NewQuery(GetType());
			q.Descend("name").Constrain("1");
			com.db4o.ObjectSet os = q.Execute();
			com.db4o.db4ounit.common.querying.CascadeOnActivate coa = (com.db4o.db4ounit.common.querying.CascadeOnActivate
				)os.Next();
			com.db4o.db4ounit.common.querying.CascadeOnActivate coa3 = coa.child.child;
			Db4oUnit.Assert.AreEqual("3", coa3.name);
			Db().Deactivate(coa, int.MaxValue);
			Db4oUnit.Assert.IsNull(coa3.name);
			Db().Activate(coa, 1);
			Db4oUnit.Assert.AreEqual("3", coa3.name);
		}
	}
}
